src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java

```diff
@@ -22,8 +26,29 @@ public ZipFileStatusService(ProcessEventRepository eventRepo, EnvelopeRepository
 
     public ZipFileStatus getStatusFor(String zipFileName) {
         return new ZipFileStatus(
-            envelopeRepo.findByZipFileName(zipFileName),
-            eventRepo.findByZipFileName(zipFileName)
+            envelopeRepo
```

Comment:
It would be best to extract those transformations into separate methods, so that this one would read easier.

Reply:
updated, let me know what you think